### PR TITLE
Add telemetry to RDS/Redshift

### PR DESCRIPTION
**Pull request by hunterwerlla**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt [hunk 1]
```diff
@@ -52,9 +56,26 @@ class SecretsManagerAuth : DatabaseAuthProvider, CoroutineScope by ApplicationTh
...
+                result = Result.Failed
+                throw e
+            } finally {
+                // We don't have access to service information here. What we do have access to is the database driver
+                // which matches up with the engine field for mysql, postgres, and redshift which is what we currently
+                // support. TODO find a more direct way to do this
+                val driver = connection.connectionPoint.databaseDriver.id
+                if (driver == "redshift") {
+                    RedshiftTelemetry.getCredentials(project, result, SecretsManager)
+                } else {
```
Comment (rli) [inline jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt:75]: nit: why not explicit RDS?

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt [hunk 2]
```diff
@@ -51,12 +54,23 @@ class IamAuth : DatabaseAuthProvider, CoroutineScope by ApplicationThreadPoolSco
...
+            var result = Result.Succeeded
+            val project = connection.runConfiguration.project
+            try {
+                val credentials = getCredentials(connection)
+                DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+            } catch (e: Throwable) {
+                result = Result.Failed
+                throw e
+            } finally {
+                // TODO find a more direct way to do this
```
Comment (rli) [inline jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt:66]: consider making this an extension so TODO is in one place

Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt:66]: Added

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt:75]: We can expand secretsmanager auth to other database engines easily, so less places to change

**Pull request merged**